### PR TITLE
Introduce repair and redeploy timers to the new agent

### PR DESCRIPTION
**Pull request by jptrindade**
# Description

Copied over some logic from the old agent to introduce repair and redeploy timers

closes *Add ticket reference here*

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [ ] Attached issue to pull request
- [x] Changelog entry
- [x] Type annotations are present
- [x] Code is clear and sufficiently documented
- [x] No (preventable) type errors (check using make mypy or make mypy-diff)
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )
- [ ] If this PR fixes a race condition in the test suite, also push the fix to the relevant stable branche(s) (see [test-fixes](https://internal.inmanta.com/development/core/tasks/build-master.html#test-fixes) for more info)

### CODE REVIEW

File: tests/agent_server/deploy/e2e/test_autostarted.py [hunk 1]
```diff
@@ -84,3 +88,159 @@ async def test_auto_deploy_no_splay(server, client, clienthelper, resource_conta
...
+    await verify_deployment_result()
+
+    # Manual change
+    resource_container.Provider.set("agent1", "key1", "another_value")
+    # Wait until repair restores the state
+    now = time.time()
+    while resource_container.Provider.get("agent1", "key1") != "value1":
+        if time.time() > now + 10:
+            raise Exception("Timeout occurred while waiting for repair run")
+        await asyncio.sleep(0.1)
```
Comment (Hugo-Inmanta) [inline tests/agent_server/deploy/e2e/test_autostarted.py:243]: The [`retry_limited`](https://github.com/inmanta/inmanta-core/blob/c4913c81e47e6dd01aaaa64a4372085d0dcf962b/tests/utils.py#L68) method can be used for this pattern.

**Review by Hugo-Inmanta — CHANGES_REQUESTED** [review]
Minor change on the test, lgtm otherwise :+1:

File: tests/agent_server/deploy/e2e/test_autostarted.py [hunk 2]
```diff
@@ -235,12 +235,12 @@ async def verify_deployment_result():
...
     # Manual change
     resource_container.Provider.set("agent1", "key1", "another_value")
+
     # Wait until repair restores the state
-    now = time.time()
-    while resource_container.Provider.get("agent1", "key1") != "value1":
-        if time.time() > now + 10:
-            raise Exception("Timeout occurred while waiting for repair run")
-        await asyncio.sleep(0.1)
+    def repaired() -> None:
```
Comment (Hugo-Inmanta) [inline tests/agent_server/deploy/e2e/test_autostarted.py:240]:
```suggestion
    def repaired() -> bool:
```

**Review by Hugo-Inmanta — CHANGES_REQUESTED** [review]

File: tests/agent_server/deploy/e2e/test_autostarted.py (hunk 2 above)
Comment (jptrindade) [reply tests/agent_server/deploy/e2e/test_autostarted.py]: yup, my bad, rushed it a little bit

**Review by sanderr — CHANGES_REQUESTED** [review]
I'll start with a general concern: we have to make sure that these interval options can be changed and that we respond to it when they are. I have yet to consider how best to address this though.

File: src/inmanta/agent/agent_new.py [hunk 3]
```diff
@@ -67,6 +69,78 @@ def __init__(
...
+        self._set_deploy_and_repair_intervals()
+
+    def _set_deploy_and_repair_intervals(self) -> None:
+        """
+        Fetch the settings related to automatic deploys and repairs from the config
+        """
+        # do regular deploys
+        self._deploy_interval = cfg.agent_deploy_interval.get()
+        deploy_splay_time = cfg.agent_deploy_splay_time.get()
+        self._deploy_splay_value = random.randint(0, deploy_splay_time)
```
[the remark this answers is not in the document]
Comment (wouterdb) [reply src/inmanta/agent/agent_new.py:84]: I think we plan to replace these timer mechanisms entirely, so yes

This implementation triggers all agents at once, so also yes

**Review by sanderr — CHANGES_REQUESTED** [review]
> I think we plan to replace these timer mechanisms entirely, so yes

@jptrindade considering Wouter's comment, I think we can leave the issue with the updating of the setting as is for now. But can you add a FIXME, noting that these settings are not currently updated (unlike the old agent) and that we should either fix it or rip this whole legacy timer out again?

**Review by sanderr — APPROVED** [review]

**Pull request closed without merging**